### ISSUE TITLE
Update to purescript 0.12.0

### ISSUE THREAD

**Issue by revazi** [paf31/purescript-book#136]
Currently I'm reading the book and working with examples, but for them to work I had to install previous version of the language. If you're not already working on example code update for the latest version I'm willing to contribute.

**Comment by vyorkin** [comment]
I see that Justin Woo has already started the work here #135.
I want to help too. As I can see there are only two chapters left (related to `purescript-react`): 8 and 10.
I can't mention him here for some reason, let's ask him in FP slack?

**Comment by vyorkin** [comment]
@revazi Just had a chat with Justin - he doesn't mind if we try to help with the remaining 2 chapters. So I'll start with chapter 8 and make a PR to his branch https://github.com/justinwoo/purescript-book/tree/0.12-psc-package.
PR for chapter 8: https://github.com/justinwoo/purescript-book/pull/1/files

**Comment by revazi** [comment]
@vyorkin Thanks for the info.  I'm currently reading book and doing exercises with previous version. Upon finishing the book(in a few days) I'll ping you and let you know how I'll be able to contribute.

**Comment by AesaKamar** [comment]
Most of my confusion was with the references to `purescript-eff`. 
As I understand now, purescript 0.12 uses `purescript-effect`?

**Comment by vyorkin** [comment]
@AesaKamar yeah, this was a major breaking change (at least most noticeable), but in most cases it's as trivial as just replacing the `Eff (foo :: FOO, bar :: BAR | eff) a` with `Effect a`

**Comment by vyorkin** [comment]
Also, the book itself needs to be updated along with the examples (after upgrading examples I suppose)
